src/Npgsql/NpgsqlCommand.cs

```diff
@@ -588,10 +589,11 @@ async Task PrepareLong()
                     for (var i = 0; i < _statements.Count; i++)
                     {
                         var statement = _statements[i];
-                        var pStatement = statement.PreparedStatement;
-                        if (pStatement?.State != PreparedState.BeingPrepared)
+                        if (!statement.IsPreparing)
                             continue;
 
+                        var pStatement = statement.PreparedStatement!;
```

Comment:
It's better to add `Debug.Assert` rather than use bangs and shoot you legs one day.

Reply:
That's a good question, I went back and forth on this too - I now think it's fine to use bangs.

In Release both are exactly the same - since the Debug.Assert gets compiled out, both will produce the same NRE in the same place. In Debug, the Debug.Assert gives you a slightly earlier assertion failure (instead of a later NRE); in some cases that can be valuable. But in example like here, the NRE would just come immediately after the assert anyway, so the effect is practically the same.

So I prefer the more terse bang over the verbose assert for these cases. What do you think?

Reply:
It's a hint for developer, but in that case I'm okay with any way. Feel free to merge as is.

Reply:
OK, thanks. I think it's a matter of people getting used to the bang, and understanding that it means exactly the same thing as Debug.Assert semantically (even if the runtime behavior is different when compiling in Debug).